tests/components/device_tracker/test_automatic.py

```diff
@@ -0,0 +1,254 @@
+"""Test the automatic device tracker platform."""
+
+import logging
+import requests
+import unittest
+from unittest.mock import patch
+
+from homeassistant.components.device_tracker.automatic import (
+    URL_AUTHORIZE, URL_VEHICLES, URL_TRIPS, setup_scanner,
+    AutomaticDeviceScanner)
+
+_LOGGER = logging.getLogger(__name__)
+
+INVALID_USERNAME = 'bob'
+VALID_USERNAME = 'jim'
+PASSWORD = 'password'
+CLIENT_ID = '12345'
+CLIENT_SECRET = '54321'
+FUEL_LEVEL = 77.2
+LATITUDE = 32.82336
+LONGITUDE = -117.23743
+ACCURACY = 8
+DISPLAY_NAME = 'My Vehicle'
+
+
+def mocked_requests(*args, **kwargs):
```

Comment:
I think `requests_mock` could potentially help shorten this, but only seen it used for basic cases so might not be ideal for you. Not critical to change